### DISCUSSION TITLE
Precision 1 and recall 0

### DISCUSSION THREAD

**Discussion by Andrei-Stefan20** [ultralytics/ultralytics#8344] [Q&A]
I wanted to inquire about this anomalous behavior, after fine-tuning on a small dataset of cells. For a class with poor representation, I find conflicting data such as precision 1 and recall 0. By definition of these metrics, these values are conflicting. From the confusion matrix, 3 classes appear completely undetected, mistaken for background or other classes, indicating an anomaly in the table. Shouldn't it actually be 0 instead of 1? Furthermore, on what basis are mAPs calculated? I cannot explain these anomalies in the evaluation table.

![confusion_matrix_normalized](https://github.com/ultralytics/ultralytics/assets/83340599/25fcc82d-af59-4abb-9f19-ce658db1aac5)
![table ](https://github.com/ultralytics/ultralytics/assets/83340599/0976ae23-45a8-4028-809c-2aba89be7ad9)

**Comment by pderrenger** [comment]
@Andrei-Stefan20 hi there! 👋 It seems like you've encountered an interesting case with the precision and recall values. Precision being 1 and recall being 0 indeed sounds contradictory, as precision = 1 typically implies some true positives, which should also affect recall.

The mAP is calculated based on the area under the precision-recall curve, so if the precision and recall values are not accurate, this might affect the mAP calculation.

To better understand what might be happening, could you please share a bit more about your dataset and the specific command you used for training and validation? Sometimes, if a class has very few examples, the metrics can behave unexpectedly.

In the meantime, you might want to double-check the following:
- Ensure that the class with poor representation has correctly labeled data.
- Verify that there's no data leakage or mislabeling that could affect the precision and recall calculations.

If you could provide the training command, it would be helpful to replicate the issue. For example:

```bash
yolo detect train data=your_dataset.yaml model=your_model.yaml
```

And for validation:

```bash
yolo detect val model=your_trained_model.pt data=your_dataset.yaml
```

Looking forward to helping you sort this out! 🛠️
